### PR TITLE
Add support for just-in-time (JIT) builds

### PR DESCRIPTION
**Pull request by diegorusso**
This PR implements https://github.com/python/pyperformance/issues/345 by specifically having the `jit` option in the configuration file.

### CODE REVIEW

**Comment by diegorusso** [comment]
Tagging @brandtbucher and @mdboom for awareness.

File: doc/benchmark.conf.sample [hunk 1]
```diff
@@ -37,6 +37,21 @@ lto = True
...
+# - no: (default) do not build the JIT or the micro-op interpreter.
+#   The new PYTHON_JIT environment variable has no effect.
+# - yes: build the JIT and enable it by default. PYTHON_JIT=0 can be used to
+#   disable it at runtime.
+# - yes-off: build the JIT, but do not enable it by default. PYTHON_JIT=1 can
+#   be used to enable it at runtime.
+# - interpreter: do not build the JIT, but do build and enable the micro-op
+#   interpreter. This is useful for those of us who find ourselves developing
+#   or debugging micro-ops (but don’t want to deal with the JIT), and is
+#   equivalent to using the -X uops or PYTHON_UOPS=1 options today.
```
Comment (brandtbucher) [inline doc/benchmark.conf.sample:51]: Outdated info:

```suggestion
#   or debugging micro-ops (but don’t want to deal with the JIT).
```

File: pyperformance/compile.py [hunk 2]
```diff
@@ -291,6 +291,8 @@ def compile(self):
             config_args.append('--with-pydebug')
         elif self.conf.lto:
             config_args.append('--with-lto')
+        if self.conf.jit and self.conf.jit in ("yes", "yes-off", "interpreter"):
```
Comment (brandtbucher) [inline pyperformance/compile.py:294]: Is there a reason you're choosing to validate this? I'd lean towards *not* validating it since any new options or defaults on the CPython side would need to be mirrored here.

```suggestion
        if self.conf.jit:
```

File: pyperformance/compile.py [hunk 3]
```diff
@@ -801,6 +803,7 @@ def getint(section, key, default=None):
         conf.directory = getfile('compile', 'bench_dir')
         conf.lto = getboolean('compile', 'lto', True)
         conf.pgo = getboolean('compile', 'pgo', True)
+        conf.jit = getstr('compile', 'jit', 'no')
```
Comment (brandtbucher) [inline pyperformance/compile.py:806]: Maybe default to the empty string, in case the upstream default gets changed at some point?

```suggestion
        conf.jit = getstr('compile', 'jit', '')
```

File: pyperformance/tests/data/find-pyperformance.py [hunk 4]
```diff
@@ -13,7 +13,7 @@
 testsroot = os.path.dirname(datadir)
 pkgroot = os.path.dirname(testsroot)
 reporoot = os.path.realpath(os.path.dirname(pkgroot))
-marker = os.path.join(reporoot, 'setup.py')
+marker = os.path.join(reporoot, 'pyproject.toml')
```
Comment (brandtbucher) [inline pyperformance/tests/data/find-pyperformance.py:16]: This seems unrelated?

File: tox.ini [hunk 5]
```diff
@@ -13,7 +13,7 @@ basepython = python3
 basepython = python3
 deps=
     sphinx
-whitelist_externals = make
+allowlist_externals = make
```
Comment (brandtbucher) [inline tox.ini:16]: Ditto. Not a bad change, but just double-checking that it doesn't really have anything to do with the change in this PR.

**Review by brandtbucher — COMMENTED** [review]
Thanks!

File: pyperformance/tests/data/find-pyperformance.py (hunk 4 above)
Comment (diegorusso) [reply pyperformance/tests/data/find-pyperformance.py:16]: I wanted to run tests and this was a blocker for running them. I left this change in a different commit so if the PR is merged without squashing this change and the next one will be in a separate commit.
If you want I can create a separate PR to include these to small changes.

Comment (diegorusso) [reply pyperformance/tests/data/find-pyperformance.py:16]: This is the commit: https://github.com/python/pyperformance/pull/351/commits/f47ab7c5d0392cb8c71b53a9dc557076e1ab7bf7

File: tox.ini (hunk 5 above)
Comment (diegorusso) [reply tox.ini:16]: This is the commit: https://github.com/python/pyperformance/pull/351/commits/f47ab7c5d0392cb8c71b53a9dc557076e1ab7bf7

File: pyperformance/tests/data/find-pyperformance.py (hunk 4 above)
Comment (diegorusso) [reply pyperformance/tests/data/find-pyperformance.py]: Dropped, I'll raise a new PR

File: tox.ini (hunk 5 above)
Comment (diegorusso) [reply tox.ini]: Dropped, I'll raise a new PR

**Comment by diegorusso** [comment]
@brandtbucher I moved the unrelated changes to this PR: https://github.com/python/pyperformance/pull/352 and I have addressed your feedback. Thanks!

**Comment by brandtbucher** [comment]
Looks good to me, thanks!
